### PR TITLE
Update marked 0.3.2 -> 0.3.6

### PR DESCRIPTION
**Pull request by islemaster**
Reverts code-dot-org/code-dot-org#19233, restoring https://github.com/code-dot-org/code-dot-org/pull/19211.  Reopened against staging-next now that we're in HOC lockdown.

In addition to upgrading marked we have to check all of our markdown for places we've used the invalid heading syntax of a hash without a space following it (`#Heading`).

Unblocked after #19274 and a subsequent i18n sync fixed this syntax in our localized level instructions.

### CODE REVIEW

**Comment by islemaster** [comment]
I _think_ I've fixed every place in our level files in `markdown_instructions` that we use the incorrect heading syntax, and I also changed a couple places in markdown files that we do so.

File: dashboard/config/scripts/levels/CSD U2 text style sample.level [hunk 1]
```diff
@@ -5,7 +5,7 @@
...
...ty</a>\r\n</details>\r\n<details>\r\n<summary>**What does the `text-align` property do?**</summary>\r\nThe `text-align` property aligns the text to the left, right, or center. You can read more at <a href=\"http://www.w3schools.com/cssref/pr_text_text-align.asp\">W3Schools - CSS text-align Property</a>\r\n</details>\r\n<details>\r\n<summary>**What does the `font-family` property do?**</summary>\r\nThe `font-family` property changes the font of the text.  It gives a general description of the type of font that should be used, but each browser might display the font slightly differently.  Read more at <a href=\"http://www.w3schools.com/cssref/pr_font_font-family.asp\">W3chools - CSS font-family Property</a>\r\n</details>\r\n</div>\r\n CSS and Style sheets\r\n\r\n* Look at the web page below and find some different types of text stylings that you don't know how to code yet.\r\n* Click on the style sheet (\"style.css\" in the files area) and look at the code inside the file.\r\n* With your partner, discuss which parts of the code you think are making the text appear differently than before.\r\n\r\nIn the next few lessons, you'll be changing and adding new styles on this page.\r\n\r\n",
```
Comment (davidsbailey) [inline dashboard/config/scripts/levels/CSD U2 text style sample.level]: intentional to remove the `#` before "CSS and Style sheets"?

Comment (islemaster) [reply dashboard/config/scripts/levels/CSD U2 text style sample.level]: Nope, that was a mistake!  I'll get it fixed.

**Comment by islemaster** [comment]
Yuck - level.yml is 4.56MB.  Feels like we should break that file up.

**Comment by islemaster** [comment]
Question - how concerned are we about the likely conflicts with levelbuilder caused by manually editing .level files in this PR?

**Comment by davidsbailey** [comment]
Though the file is large, it seems like merge conflicts on other lines in that file would be able to be auto-resolved from the command line. Probably this means that if the 3pm DTL fails, someone just has to login to levelbuilder and manually git pull.

**Comment by davidsbailey** [comment]
If that still sounds risky, you could make those changes via levelbuilder instead, then merge this PR after the noon levelbuilder scoop and DTS.

**Comment by davidsbailey** [comment]
Oops, I totally misread your comment. There may be some merge conflicts. perhaps it's best to sync latest staging after the noon levelbuilder scoop/merge, fix up any conflicts, merge this to staging and then quickly "hotfix" it to levelbuilder.

**Comment by islemaster** [comment]
Nice catch!  I forgot to look at localization files, and the bad markdown syntax still exists throughout markdown_instructions.en.yml.  We'll have to step back and figure out the right approach here.

**Comment by islemaster** [comment]
This PR is no longer blocked by https://github.com/code-dot-org/code-dot-org/pull/19274 - that change has been merged and we had a fresh i18n pull yesterday, so our localized content should have the markdown syntax fixes as well.

image

**Comment by islemaster** [comment]
Removed all but the dependency change (the rest should have been in the other PR) and changed base to `staging-next`.

**Pull request merged**